temba/msgs/models.py

```diff
@@ -1400,9 +1433,24 @@ def evaluate_template(cls, text, context, org=None, url_encode=False, partial_va
         return evaluate_template(text, context, url_encode, partial_vars)
 
     @classmethod
+    def get_outgoing_metadata(cls, metadata, expressions_context, contact, org, channel):
+        quick_replies = metadata.get('quick_replies', None)
+        if quick_replies:
+            for counter, reply in enumerate(quick_replies):
+                (text, errors) = Msg.evaluate_template(reply, expressions_context, org=org)
+                if text:
+                    if contact.is_test:
+                        quick_replies[counter] = text
+                    else:
+                        channel_type = Channel.get_type_from_code(channel.channel_type)
+                        quick_replies[counter] = text[:channel_type.quick_reply_text_size]
+
```

Comment:
Like to avoid in-place changes, instead create a new list and return that. Also, looks like this classmethod is used only once. Is that intentional?